### PR TITLE
feat(auth): create base FormField component

### LINKED ISSUE #5794: LoginPage - Phase 2a - Create Base FormField (UI only)
### Problem Statement
Inconsistent field markup and a11y in LoginPage; create a base FormField without validator coupling to unblock downstream fields.

### What Needs to Be Done
Implement the work described in the Implementation Approach and meet the Acceptance Criteria.

### Implementation Approach
- Follow Option B directory structure: src/components/Auth/**
- Keep props/interfaces in src/types/Auth/** with brief TSDoc
- Adhere to the reusable components guidance
- Avoid coupling Phase 2a UI with validators; defer to Phase 2b

### Files to Modify (initial target paths)
- [ ] src/components/Auth/FormField/FormField.tsx
- [ ] src/components/Auth/FormField/FormField.spec.tsx
- [ ] src/types/Auth/FormField/interface.ts

### Acceptance Criteria
- [ ] No imports from src/utils/validators/**
- [ ] Props accept precomputed error?: string | null
- [ ] A11y: aria-invalid/aria-describedby with error feedback

### Testing Requirements
- [ ] Tests cover render, error display, and accessibility attributes

### Dependencies
- See dependency comments (Blocked by/Blocks) on this issue

### Phase
Phase 2a (UI Foundation)

## Implementation Plan
- https://github.com/PalisadoesFoundation/talawa-admin/issues/5761

### Metadata
- Assignee: palisadoes-bot
- Labels: refactor

### Example Starter Code

```ts
// src/types/Auth/FormField/interface.ts
import type { ChangeEvent, FocusEvent } from 'react';
export interface FormFieldProps {
  label?: string; name: string; type?: 'text'|'email'|'password'|'tel'; value: string;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void; onBlur?: (e: FocusEvent<HTMLInputElement>) => void;
  placeholder?: string; required?: boolean; disabled?: boolean; testId?: string; error?: string | null;
}
```

```tsx
// src/components/Auth/FormField/FormField.tsx
import React from 'react'; import { Form } from 'react-bootstrap';
import type { FormFieldProps } from '@/types/Auth/FormField/interface';
export const FormField: React.FC<FormFieldProps> = (p) => {
  const errorId = p.error ? `${p.name}-error` : undefined;
  return (
    <Form.Group className="mb-3" controlId={p.name}>
      {p.label && <Form.Label>{p.label}{p.required && <span className="text-danger"> *</span>}</Form.Label>}
      <Form.Control type={p.type ?? 'text'} name={p.name} value={p.value} onChange={p.onChange} onBlur={p.onBlur}
        placeholder={p.placeholder} disabled={p.disabled} isInvalid={!!p.error}
        aria-invalid={!!p.error} aria-describedby={errorId} data-testid={p.testId}/>
      {p.error && <Form.Control.Feedback type="invalid" id={errorId} className="d-block">{p.error}</Form.Control.Feedback>}
    </Form.Group>
  );
};
```

## Implementation Plan
- https://github.com/PalisadoesFoundation/talawa-admin/issues/5761

_This issue supersedes the previous version to include code snippets._

### PR DESCRIPTION
**Pull request by NavyasreeBalu**
## What kind of change does this PR introduce?

Feature, Refactor

## Issue Number

Fixes #5794

## Snapshots / Videos

N/A. This is a foundational UI component that is not yet integrated into any screen.

## If relevant, did you update the documentation?

N/A. All documentation is included as TSDoc comments in the code.

## Summary

This PR introduces a foundational, reusable `FormField` component as part of **Phase 2a** of the LoginPage refactoring plan (#5761).

The purpose of this component is to create a consistent, accessible, and “dumb” base for all future form inputs. It is intentionally decoupled from validation logic and simply displays a pre-computed error message if one is passed via props.

This resolves the issue of inconsistent form field markup in the existing `LoginPage`.

## Does this PR introduce a breaking change?

No. This is an additive change and does not modify any existing code.

### CodeRabbit AI Review

- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI  
- [x] I have implemented or provided justification for each non-critical suggestion  
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented  

### Test Coverage

- [x] I have written tests for all new changes/features  
- [x] I have verified that test coverage meets or exceeds 95% (100% for the new component)  
- [x] I have run the test suite locally and all tests pass  

## Other Information

- Creates the `FormField` component in `src/components/Auth/FormField/`
- Defines the `FormFieldProps` in `src/types/Auth/FormField/`
- Implements a comprehensive test suite verifying:
  - Rendering
  - Error states
  - Accessibility (`aria-invalid`, `aria-describedby`)

## Contributing Guide

- [x] Yes, I have read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)

## Summary by CodeRabbit

* **New Features**
  * Added a new form field component with integrated validation, error display, accessibility features (aria labels, error linking), and support for multiple input types.

* **Tests**
  * Added comprehensive test coverage for the form field component, including rendering variants, error handling, accessibility requirements, and user interaction scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by NavyasreeBalu** [comment]
> Exclude package.json from the PR

 @palisadoes My bad on the package.json, that wasn't supposed to be in there.

Fixed in c8098376f2 and also worked on the CodeRabbit issues in a4a3e99175. PR is now clean and focused just on the FormField component per #5794.

Thanks for catching that!

**Comment by NavyasreeBalu** [comment]
@palisadoes  I'm not able to understand this E2E test failure - I don't think it's caused by my FormField code?

The error seems to be about devcontainer configuration ("version": "default" not found), but I only added new FormField component files. Can you guide me on whether this is related to my changes or if it's a separate infrastructure issue?

All my FormField tests are passing with 100% coverage, so I'm not sure if I need to fix something

**Comment by chauhan-varun** [comment]
hi @NavyasreeBalu its not caused by your FormField code. 
The current failing test is due to an existing known issue, which is already open here: 
- https://github.com/PalisadoesFoundation/talawa-admin/issues/5875

**Comment by NavyasreeBalu** [comment]
> hi @NavyasreeBalu its not caused by your FormField code. The current failing test is due to an existing known issue, which is already open here:
> 
> * [E2E Tests: Flaky failures in event.cy.ts due to timeouts and sequential dependencies #5875](https://github.com/PalisadoesFoundation/talawa-admin/issues/5875)

@chauhan-varun Got it, thanks for confirming! Let me know if there's
anything I can do to help move this PR forward.

**Review by palisadoes — APPROVED** [review]

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please ensure code rabbit approves your changes

File: src/types/Auth/FormField/interface.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  type?: 'text' | 'email' | 'password' | 'tel';
+  value: string;
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  required?: boolean;
+  disabled?: boolean;
+  testId?: string;
+  error?: string | null;
+}
```
[the remark this answers is not in the document]
Comment (palisadoes) [reply src/types/Auth/FormField/interface.ts:19]: @NavyasreeBalu Please address this

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all resolved items and include the following points:

1. **Test Coverage & Quality Validation:**
    1. Validate whether any updated test files:
       1. Cover all use cases including edge cases, error scenarios, and success paths
       2. Have valid, properly structured tests following Jest and React Testing Library best practices
       3. Provide 100% test code coverage for all modified/added functions, components, and branches
       4. Include proper mocking of GraphQL queries/mutations, external dependencies, and API calls
       5. Test both user interactions and state changes comprehensively
       6. Require any additional tests for uncovered scenarios
    1. For each modified file with logic or UI, verify corresponding test files exist and are updated:
        1. Check that test files follow naming conventions (_.test.tsx, _.test.ts, _.spec.tsx, _.spec.ts)
        1. Validate that tests don't include disabled tests (it.skip, describe.skip) unless explicitly documented
        1. Ensure mock data cleanup is performed (as required by pre-commit hooks)

1. **Code Quality & Repository Standards:**
    1. Verify TypeScript compliance:
         1. No use of `any` type without explicit justification
         2. Proper interface/type definitions for all props, state, and function parameters
         3. No TypeScript errors or warnings
    1. Validate GraphQL implementation:
         1. Queries/mutations are properly typed and follow existing patterns in src/GraphQl/
         2. No duplicate or redundant queries (check for pagination inconsistencies)
         3. Error handling is implemented for all GraphQL operations
    1. Check React component best practices:
         1. Functional components with proper hooks usage
         2. Material-UI component patterns are followed consistently
         3. No prop drilling; proper use of context when needed
         4. Proper cleanup in useEffect hooks
    1. Verify internationalization (i18n):
         1. All user-facing strings use i18next translation keys
         2. New translation keys are added to locale files
         3. No hardcoded English text in UI components
    1. Validate accessibility:
         1. Proper ARIA labels and roles
         2. Keyboard navigation support
         3. Screen reader compatibility
  
1. **Documentation & Code Comments:**
   1. Verify auto-generated documentation is updated (as enforced by Check-AutoDocs job)
   2. Check for adequate JSDoc comments on public functions and complex logic
   3. Ensure README.md or related docs are updated if new features are added
   4. Validate that Table of Contents is updated (as required by pre-commit hooks)

1. **Identify Issues:**
   1. Lines of code that are missing tests (list specific line numbers)
   2. Any unnecessary files that have been submitted (build artifacts, node_modules, .env files, IDE configs)
   3. Unused imports or dead code (as checked by knip)
   4. Console.log statements or debug code left in production files
   5. Files that should be excluded from the PR scope

1. **PR Requirements Validation:**
   1. Verify the PR meets the requirements of the linked issue:
      1. All acceptance criteria are addressed
      2. No out-of-scope changes are included
   2. Validate compliance with all previously suggested improvements and comments:
      1. All review comments have been addressed
      2. Requested changes have been implemented correctly
   3. Check PR metadata:
      1. Proper title and description
      2. Linked to appropriate issue(s)
      3. Target branch is correct (typically `develop`)
   4. Verify no sensitive information is exposed (as checked by Check-Sensitive-Files job)

1. **CI/CD Pipeline Compatibility:**
   1. Validate that code changes won't break any of the 17+ CI jobs
   2. Ensure pre-commit hook requirements are met:
      1. Code formatting (Prettier)
      2. Linting (ESLint)
      3. Type checking (TypeScript)
   3. Check that changes don't introduce performance regressions for the 12-shard test execution

1. **Security & Best Practices:**
   1. No security vulnerabilities introduced
   2. No exposed API keys, tokens, or credentials
   3. Proper input validation and sanitization
   4. No SQL injection or XSS vulnerabilities in any data handling

1. **Implementation Correctness:**
   1. For the specific implementation file(s) being tested:
      1. Verify the implementation logic is correct and free of bugs
      1. Validate that queries target the correct database tables/entities
      1. Check that foreign key relationships are correctly enforced
      1. Confirm business logic constraints are properly implemented
      1. Any bugs, logic errors, or incorrect implementations discovered in the file(s) under test MUST be fixed in this PR before approval

1. **Review Decision:**
   1. When critical issues are found, clearly state that changes are required in your response but DO NOT append the [approve] tag.
   1. Only append the [approve] tag when ALL criteria are satisfied and zero changes are required:
      1. Complete test coverage (≥95%)
      2. No code quality issues, security concerns, or bugs
      3. Full implementation of issue requirements
      4. No bugs or logic errors in the implementation file(s) under test
      5. Compliance with all previous feedback
      6. No unnecessary files

1. **Approvals:**
   1. Approve the PR by appending the text "APPROVED" below your response only if:
   2. All criteria above are fully satisfied
   3. The implementation file(s) under test are correct and bug-free
   4. No changes, even trivial ones, are required

1. **Focus Areas:**
   1. Prioritize substantive issues over minor stylistic suggestions
   2. Focus on issues that would cause CI pipeline failures or production bugs
   3. Highlight any deviations from established codebase patterns

**Note:** Test-focused PRs should maintain clear scope boundaries. However, this does NOT exempt bugs in the specific file(s) being tested - those must be fixed in the same PR. Only bugs in unrelated files may be deferred to separate issues.
